### PR TITLE
chore: Simplify views/base

### PR DESCRIPTION
**Pull request by sebastianliebscher**
### SUMMARY
Simplifies views/base.py to make code more readable / maintainable by using 
- removed unused function parameters
- dict comprehension instead of for loop
- X | Y type hints (via `pyupgrade`)

### TESTING INSTRUCTIONS
- local unit / integration tests + GitHub CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/base.py [hunk 1]
```diff
@@ -139,15 +141,11 @@ def get_error_msg() -> str:
 
 
 def json_error_response(
-    msg: Optional[str] = None,
+    msg: str | None = None,
     status: int = 500,
-    payload: Optional[dict[str, Any]] = None,
-    link: Optional[str] = None,
```
Comment (sebastianliebscher) [inline superset/views/base.py:145]: `link` parameter is not used in the codebase at all

File: superset/views/base.py [hunk 2]
```diff
@@ -192,15 +189,15 @@ def generate_download_headers(
...
-    new_target: Optional[str] = None,
+    new_target: str | None = None,
 ) -> Callable[[Callable[..., FlaskResponse]], Callable[..., FlaskResponse]]:
     """
     A decorator to set an API endpoint from SupersetView has deprecated.
     Issues a log warning
     """
 
     def _deprecated(f: Callable[..., FlaskResponse]) -> Callable[..., FlaskResponse]:
-        def wraps(self: "BaseSupersetView", *args: Any, **kwargs: Any) -> FlaskResponse:
```
Comment (sebastianliebscher) [inline superset/views/base.py:203]: this was auto-changed by pre-commit `pyupgrade`

File: superset/views/base.py (hunk 1 above)
Comment (john-bodley) [reply superset/views/base.py:145]: @sebastianliebscher it seems like the `payload` argument isn't use in the codebase either.

File: superset/views/base.py [hunk 3]
```diff
@@ -682,18 +674,14 @@ def apply(self, query: Query, value: Any) -> Query:
 
 
 class CsvResponse(Response):
-    """
-    Override Response to take into account csv encoding from config.py
-    """
+    """Override Response to take into account csv encoding from config.py"""
```
Comment (john-bodley) [inline superset/views/base.py:677]: Were these comments manually or automatically changed?

**Review by john-bodley — COMMENTED** [review]
Thanks @sebastianliebscher for the cleanup. I left a few minor comments, but overall this LGTM.

File: superset/views/base.py (hunk 1 above)
Comment (sebastianliebscher) [reply superset/views/base.py:145]: there is a single usage here https://github.com/apache/superset/blob/d95c200e6931e2bc14b0448e3da265d8ebedf249/superset/views/core.py#L166

File: superset/views/base.py (hunk 3 above)
Comment (sebastianliebscher) [reply superset/views/base.py:677]: Manually. If that change is not desired, I'll revert it

Comment (john-bodley) [reply superset/views/base.py:677]: @sebastianliebscher could you please revert these docstring changes.

Comment (sebastianliebscher) [reply superset/views/base.py]: @john-bodley Done.

**Comment by sebastianliebscher** [comment]
The test that is failing is unrelated. There is another PR to address this https://github.com/apache/superset/pull/25958

**Review by john-bodley — APPROVED** [review]

**Pull request merged**